mypy/checker.py

```diff
@@ -1736,7 +1736,7 @@ def check_simple_assignment(self, lvalue_type: Type, rvalue: Expression,
             # '...' is always a valid initializer in a stub.
             return AnyType()
         else:
-            rvalue_type = self.expr_checker.accept(rvalue, lvalue_type)
+            rvalue_type = self.expr_checker.accept(rvalue, lvalue_type, disallow_any=False)
```

Comment:
Yeah, that's a good point. I will change that to `disallow_any=lvalue_type is None`. It seems like `lvalue_type` is the type from an explicit type annotation.

---

As for not allowing `Any` expressions when assigning to a variable with an explicit type annotation, here are my thoughts.

People are going to be calling untyped functions from their code. `--disallow-any=expr` needs to allow for this, otherwise no one will use it.
In my mind, there are two good ways to do it: explicit cast (call to function `cast`) and implicit cast (assignment to a variable with an explicit type annotation). 
The advantage of an explicit cast is that you can't do it accidentally.
However, the explicit cast syntax makes the code less readable.
Compare with an implicit cast:

```py
# explicit cast
result = cast(float, some_long_method_name(that, (has, many), complicated=params))
# implicit cast
result = some_long_method_name(that, (has, many), complicated=params)  # type: float
```

A big disadvantage of an implicit cast is that someone can use it accidentally. They won't look at the definition of the function they are calling and just put a type there thinking that mypy will complain if the type is wrong.

This PR has both implicit and explicit cast implemented because I think it's worth having both.
But I would be interested in hearing your reasoning for this!